device_flow_test.py

```diff
@@ -0,0 +1,76 @@
+import os
+import pathlib
+
+import simple_ado
+import argparse
+import logging
+from auth_helper import AuthHelper
+from simple_ado import ADOException
+from simple_ado.exceptions import ADOHTTPException
+
+def main():
+    logger = logging.getLogger("test")
+    project_id = os.environ["SIMPLE_ADO_PROJECT_ID"]
```

Comment:
Where are these coming from?

Reply:
Do you mean where are the os environment variables coming from? I'm not following.

Reply:
Correct. I'd like to see a test that's reproducible.

Reply:
Hmm... ok, so the challenge is I'm not sure what I can share there. All of my tests are using values I've setup for internal usage at Microsoft. Getting this to work for project/repo/org etc. should be easy enough, but the authentication layer requires appId and a variety of other values that I'm not sure can be publicly provided. Let me look into that... are you by chance Microsoft-based? We could discuss this in a teams chat and I could show you there. Then we can figure out how we want to address it for the public.

Reply:
Ok, I think I can do this, based on conversations I've had internally, nothing of these are considered secrets. If you can approve PR, I can re-pull, move this all to a test case, and add it there.